### DISCUSSION TITLE
mastodon saying I did not set a S3 region

### DISCUSSION THREAD

**Discussion by bobby365** [mastodon/mastodon#29886] [Q&A] [answered]
In sidekiq I see Aws::Errors::MissingRegionError: No region was provided. Configure the `:region` option or export the region name to ENV['AWS_REGION']

Well I have 

S3_PROTOCOL=https
S3_ENDPOINT=https://s3.sexypokemon.xyz
S3_HOSTNAME=s3.sexypokemon.xyz
S3_BUCKET=mastodon
S3_REGION=EU1

It works because I can upload media to the server but over federation it does not work

**Comment by StrangeMinds1960** [comment]
This is a list of all the Amazon S3 regions, there's no EU1 in there.  You'll need to check your S3 settings in AWS to see which one you need to use

https://docs.aws.amazon.com/general/latest/gr/s3.html

**Reply by bobby365** [reply]
My minio S3 is set to have a region EU1. So yes in my S3 there is EU1

**Reply by bobby365** [reply]
I set my minio server region to eu-central-1 and then set my S3 region on my server to the same thing and it’s still complaining

**Comment by StrangeMinds1960** [comment]
Was that your full configuration you posted?  There's thing missing that I think should be there, including authentication between Mastodon and Minio

This is my configuration in full

S3_ENABLED=true
S3_BUCKET=strangeminds
AWS_ACCESS_KEY_ID=**********************
AWS_SECRET_ACCESS_KEY=*********************
S3_REGION=eu-west-2
S3_PROTOCOL=https
S3_HOSTNAME=s3-eu-west-2.amazonaws.com
S3_ALIAS_HOST=assets.strangeminds.social

**Reply by bobby365** [reply]
```
S3_ENABLED=true
# S3_OVERRIDE_PATH_STYLE=https://s3.sexypokemon.xyz/mastodon/
S3_PROTOCOL=https
S3_ENDPOINT=http://IP:PORT
S3_HOSTNAME=s3.sexypokemon.xyz
S3_BUCKET=mastodon
AWS_REGION=eu-central-1
S3_REGION=eu-central-1
AWS_ACCESS_KEY_ID=**********************
AWS_SECRET_ACCESS_KEY=**********************
```

**Comment by StrangeMinds1960** [comment]
Have you seen this?  Not sure if it'll help or not

https://docs.joinmastodon.org/admin/optional/object-storage/#minio

**Reply by bobby365** [reply]
I did and it did nothing

**Comment by bobby365** [comment, marked as answer]
I give up. I'm just going to pay for S3 hosting
